profiler/test/Datadog.Profiler.IntegrationTests/GarbageCollections/GarbageCollectorCpuTimeTest.cs

```diff
@@ -0,0 +1,173 @@
+// <copyright file="GarbageCollectorCpuTime.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+// </copyright>
+
+using Datadog.Profiler.IntegrationTests.Helpers;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Profiler.IntegrationTests.GarbageCollections
+{
+    // Tests are done only for .NET 5+ because GC threads got names starting .NET 5
+    // And this feature only work in case of GC server setting (not workstation)
+    public class GarbageCollectorCpuTimeTest
+    {
+        private const string ScenarioGenerics = "--scenario 12";
+        private const string GcFrame = "|lm: |ns: |ct: |fn:Garbage Collector";
+
+        private readonly ITestOutputHelper _output;
+
+        public GarbageCollectorCpuTimeTest(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [TestAppFact("Samples.Computer01", new[] { "net6.0", "net7.0" })]
+        public void CheckCpuTimeForGcThreadsIsReported(string appName, string framework, string appAssembly)
+        {
+            var runner = new TestApplicationRunner(appName, framework, appAssembly, _output, commandLine: ScenarioGenerics);
+            runner.Environment.SetVariable(EnvironmentVariables.GcThreadsCpuTimeEnabled, "1");
+
+            // Enable walltime and check GC frame does not have a value for this column
+            runner.Environment.SetVariable(EnvironmentVariables.WallTimeProfilerEnabled, "1");
+            // enable cputime profiler to ensure we get cpu time for GC threads
+            runner.Environment.SetVariable(EnvironmentVariables.CpuProfilerEnabled, "1");
+            // Enable GC Server
+            runner.Environment.SetVariable("COMPlus_gcServer", "1");
+
+            using var agent = MockDatadogAgent.CreateHttpAgent(_output);
+            runner.Run(agent);
+            Assert.True(agent.NbCallsOnProfilingEndpoint > 0);
+
+            var gcStackTrace = new StackTrace(new StackFrame(GcFrame));
+
+            SamplesHelper.GetSamples(runner.Environment.PprofDir).Should()
+                // match the GC stacktrace and check that the waltime value is 0 and the cpu value is not 0
+                .Contain(sample => sample.StackTrace.Equals(gcStackTrace) && sample.Values[0] == 0 && sample.Values[1] != 0);
```

Comment:
It is kind of risky to assumes that wall time is the first value and cpu is the second. We might change the order in the future. Maybe just checking the GC callstack is enough.

Reply:
By construction we know it's that way. How can I be sure that the CPU value is filled ?